gtsfm/pose_slam/pose_slam.py

```diff
@@ -20,6 +23,106 @@
 
 
 class PoseSlam:
+    """Initializes using PoseSLAM given relative pose priors."""
+
+    @staticmethod
+    def angle(R1, R2):
+        """Calculate angle between two rotations, in degrees."""
+        return np.degrees(np.linalg.norm(R1.logmap(R2)))
+
+    @staticmethod
+    def difference(P1, P2):
+        """Calculate the translation and angle differences of two poses.
+        P1, P2: Pose3
+        Return:
+            distance: translation difference
+            angle: angular difference
+        """
+        # TODO(frank): clean this up
+        t1 = P1.translation()
+        t2 = P2.translation()
+        R1 = P1.rotation()
+        R2 = P2.rotation()
+        R1_2 = R1.compose(R2.inverse())
+        t1_ = R1_2.rotate(t2)
+        # t1_2 = t1 - R1_2*t2
+        distance = np.linalg.norm(t1 - t1_)
+        angle_ = PoseSlam.angle(R1, R2)
```

Comment:
confusing without the reference frame.

Reply:
Will remove, this is confusing code, I agree